plugins/kubernetes/app/models/kubernetes/service.rb

```diff
@@ -0,0 +1,95 @@
+module Kubernetes
+  # Convenience wrapper around the 'Service' object we can get from the K8S API
+  class Service
+    attr_reader :deploy_group, :role, :name, :ip_address, :port_info
+
+    def initialize(deploy_group: nil, role: nil)
+      @deploy_group = deploy_group
+      @role = role
+      @port_info = []
+      @name = role.try(:service_name)
+    end
+
+    def namespace
+      deploy_group.kubernetes_namespace
+    end
+
+    def running?
+      service_object.present?
+    end
+
+    def proxy_address
+      "#{example_node_ip}:#{node_ports.first}" if running?
+    end
+
+    def example_node_ip
```

Comment:
Yeah. In Kubernetes, any of the nodes acts as a proxy/load-balancer to any of the other nodes. There's no special server that has that function.  So if you want to send traffic to the pods that are part of a service, and the service has exposed a `NodePort`, you send a request to that port on any of the nodes (aka minions).

I created this method for the `proxy_address` method, which is nice to display in the UI so that a user in dev mode can see what the address is for the app.